src/components/JazzIcon.tsx

```diff
@@ -0,0 +1,32 @@
+import { Box } from '@chakra-ui/react';
+import Jazzicon from '@metamask/jazzicon';
+import { type FunctionComponent, useRef, useEffect } from 'react';
+
+export type JazzIconProps = {
+  address: string;
+  size: number;
+};
+
+export const JazzIcon: FunctionComponent<JazzIconProps> = ({
+  address,
+  size,
+}) => {
+  const ref = useRef<HTMLDivElement>();
+  const addr = address.trim().slice(2, 10);
+  const seed = parseInt(addr, 16);
+
+  useEffect(() => {
+    if (ref?.current) {
+      ref.current.innerHTML = '';
+      ref.current.appendChild(Jazzicon(size, seed));
+    }
+  }, [ref, seed, size]);
+
+  return (
+    <Box
+      data-testid="jazzicon"
+      ref={ref as React.MutableRefObject<HTMLDivElement>}
```

Comment:
Why is the type cast necessary?

Reply:
yup, it can be remove

Reply:
the type is to ignore the type script error

Reply:
Seems like there is an underlying issue then? You shouldn't need a type cast to set a ref.

Reply:
if you take a look the jazz icon is returning the div, instead of a react component, hence you have to using appendChild to include the jazzicon, therefore we have to cast ref as htmldivelement, otherwise, you will not able to using `appendChild` method of the ref.current

if this is really a big blocker, please let us know how to fix it

Reply:
```ts
const ref = useRef<HTMLDivElement>(null);

// ...

ref={ref}
```

Reply:
done